torchvision/transforms/functional_tensor.py

```diff
@@ -586,8 +586,8 @@ def resize(img: Tensor, size: List[int], interpolation: int = 2) -> Tensor:
         else:
             size_w = int(size_h * w / h)
 
-    if (w <= h and w == size_w) or (h <= w and h == size_h):
-        return img
```

Comment:
@vfdev-5 Could you explain what your intention for this guarding clause was? Is this just to detect the case if no resizing needs to happen? If yes, why not simply do

```python
if (w,h) == (size_w, size_h):
    return 
```

Reply:
I think his intention is that (according to the function docstring), if only one number is given as size, we resize to match the smaller dimension to that number. The check there is for the case when the smaller dimension matches the number given and no resizing needs to happen. But it doesn't mean that both dims need to match

Reply:
Sorry for delayed reply! Yes, this function should mimic PIL implementation : https://github.com/pytorch/vision/blob/master/torchvision/transforms/functional_pil.py#L324 where definitely this condition is applied in case when `if isinstance(size, int) or len(size) == 1`. Thanks for the fix @ag14774 !